modules/ezcontent_preview/src/Controller/PreviewView.php

```diff
@@ -5,30 +5,43 @@
 use Drupal\Core\Controller\ControllerBase;
 use Drupal\node\NodeInterface;
 use Drupal\ezcontent_preview\Utils;
+use Drupal\Core\Routing\TrustedRedirectResponse; 
 
 
 /**
  * Preview the content
  */
 class PreviewView extends ControllerBase {
 
-  public function preview(NodeInterface $node = NULL) {
-    $urlUtils = new Utils();
-    $url = $urlUtils->buildUrl($node);
-    if ($url) {
-      $output = '<iframe class="decoupled-content--preview" src="' . $url->toString() . '"></iframe>';
-      return array(
-        '#type' => 'markup',
-        '#allowed_tags' => ['iframe'],
-        '#markup' => $output,
-        '#attached' => [
-          'library' => [
-            'ezcontent_preview/global',
-          ],
-        ],
-      );
-    }
+  public function preview(NodeInterface $node = NULL, $preview_type) {
+
+    if ($preview_type) {
+      $decoupledRoutes = \Drupal::entityTypeManager()->getStorage('ezcontent_preview')->load($preview_type);
```

Comment:
Can we inject the entity type manager object.

Reply:
added DI for it.